### PR TITLE
use utils in knative/pkg

### PR DESCRIPTION
**Pull request by runzexia**
Signed-off-by: runzexia <root@junwuhui.cn>
knative/eventing#2884

### CODE REVIEW

File: pkg/reconciler/kafka/kafka.go [hunk 1]
```diff
@@ -21,19 +21,19 @@ import (
...
+	kafkav1beta1 "knative.dev/eventing-contrib/kafka/source/pkg/apis/sources/v1beta1"
 )
 
 const (
 	defaultKafkaLagThreshold = 10
 )
 
 func GenerateScaleTargetName(src *kafkav1beta1.KafkaSource) string {
-	return eventingutils.GenerateFixedName(src, fmt.Sprintf("kafkasource-%s", src.Name))
+	return kmeta.ChildName(fmt.Sprintf("kafkasource-%s-", src.Name), string(src.GetUID()))
```
Comment (zroubalik) [inline pkg/reconciler/kafka/kafka.go:36]: The Kafka Source is still using `utils.GenerateFixedName()` see: https://github.com/knative/eventing-contrib/blob/315af7ccf3a0d519c04e4bc9407feb8f8a125c43/kafka/source/pkg/reconciler/source/resources/receive_adapter.go#L84
So we should always use the same, to target the same deployment.
Am I missing anything?

File: pkg/reconciler/source/source.go [hunk 2]
```diff
@@ -153,6 +157,16 @@ func (r *Reconciler) reconcileAwsSqsSource(ctx context.Context, src *awssqsv1alp
 func (r *Reconciler) reconcileScaledObject(ctx context.Context, expectedScaledObject *kedav1alpha1.ScaledObject, obj metav1.Object) error {
 	scaledObject, err := r.kedaClient.KedaV1alpha1().ScaledObjects(expectedScaledObject.Namespace).Get(ctx, expectedScaledObject.Name, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
+		// Issue eventing#2842: Adater deployment name uses kmeta.ChildName. If a deployment by the previous name pattern is found, it should
+		// be deleted. This might cause temporary downtime.
+		// only kafka use GenerateFixedName, so we only try kafka.
+		if deprecatedName := utils.GenerateFixedName(obj, fmt.Sprintf("kafkasource-%s", obj.GetName())); deprecatedName != expectedScaledObject.Name {
+			if err := r.kubeClient.AppsV1().Deployments(obj.GetNamespace()).Delete(ctx, deprecatedName, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
```
Comment (zroubalik) [inline pkg/reconciler/source/source.go:164]: I don't think we should delete or modify the Deployment (Adapter) from this controller. This should be handled by Kafka Source Controller

**Review by zroubalik — CHANGES_REQUESTED** [review]

File: pkg/reconciler/source/source.go (hunk 2 above)
Comment (runzexia) [reply pkg/reconciler/source/source.go:164]: ok

File: pkg/reconciler/source/source.go [hunk 3]
```diff
@@ -20,6 +20,10 @@ import (
 	"context"
 	"fmt"
 
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/eventing/pkg/utils"
+	"knative.dev/pkg/controller"
+
```
Comment (mattmoor) [inline pkg/reconciler/source/source.go:26]: Format Go code:
```suggestion

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: pkg/reconciler/kafka/kafka.go (hunk 1 above)
Comment (runzexia) [reply pkg/reconciler/kafka/kafka.go:36]: create https://github.com/knative/eventing-contrib/pull/1619/ to handle Kafka Source

Comment (zroubalik) [reply pkg/reconciler/kafka/kafka.go:36]: OK, we should wait for that PR to be merged in eventing-contrib

**Review by n3wscott — COMMENTED** [review]
/lgtm
/approve

**Comment by runzexia** [comment]
@n3wscott @zroubalik 
Should we pin kafkasource to https://github.com/knative-sandbox/eventing-kafka?
If it's need, I can help

**Comment by zroubalik** [comment]
> @n3wscott @zroubalik
> Should we pin kafkasource to https://github.com/knative-sandbox/eventing-kafka?
> If it's need, I can help

@runzexia is that repo the proper source of KafkaSource now? I haven't followed the Kafka related changes recently.

**Comment by runzexia** [comment]
> > @n3wscott @zroubalik
> > Should we pin kafkasource to https://github.com/knative-sandbox/eventing-kafka?
> > If it's need, I can help
> 
> @runzexia is that repo the proper source of KafkaSource now? I haven't followed the Kafka related changes recently.

https://github.com/knative/eventing-contrib/pull/1619#issuecomment-707561521
yes, as mentioned

**Comment by zroubalik** [comment]
> > @runzexia is that repo the proper source of KafkaSource now? I haven't followed the Kafka related changes recently.
> 
> [knative/eventing-contrib#1619 (comment)](https://github.com/knative/eventing-contrib/pull/1619#issuecomment-707561521)
> yes, as mentioned

@runzexia thanks, we should change that as well then 👍  Would be great if you can look at that.

**Comment by runzexia** [comment]
> @runzexia thanks, we should change that as well then 👍 Would be great if you can look at that.

ok will try it ☺️

**Pull request merged**